code/modules/heavy_vehicle/mech_damage.dm

```diff
@@ -37,17 +37,31 @@
 	maxHealth = body.mech_health
 	health = maxHealth-(getFireLoss()+getBruteLoss())
 
-/mob/living/heavy_vehicle/adjustFireLoss(var/amount)
-	var/obj/item/mech_component/MC = pick(list(arms, legs, body, head))
-	if(MC)
-		MC.take_burn_damage(amount)
-		MC.update_health()
-
-/mob/living/heavy_vehicle/adjustBruteLoss(var/amount)
-	var/obj/item/mech_component/MC = pick(list(arms, legs, body, head))
-	if(MC)
-		MC.take_brute_damage(amount)
-		MC.update_health()
+/mob/living/heavy_vehicle/adjustFireLoss(var/amount, var/obj/item/mech_component/C)
+	if(C)
+		C.take_brute_damage(amount)
+		C.update_health()
+	else
+		var/list/components = list(body, arms, legs, head)
+		components = shuffle(components)
+		for(var/obj/item/mech_component/MC in components)
+			if(MC)
```

Comment:
Do we actually need to check if MC exists here? If it doesn't exist, something has gone horribly wrong with the components list and we should probably runtime.

Reply:
We don't have to check.